tests/unit/internal-session-test.js

```diff
@@ -776,6 +776,28 @@ describe('InternalSession', () => {
               done();
             });
           });
+
+          it('it does not trigger the "sessionInvalidationFailed" event', function(done) {
+            let triggered = false;
+            sinon.stub(session, '_lookupAuthenticator').returns(authenticator);
+            session.one('sessionInvalidationFailed', () => (triggered = true));
+
+            session.invalidate().then(() => {
+              expect(triggered).to.be.false;
+              done();
+            });
+          });
+
+          it('it returns with a resolved Promise', function(done) {
+            sinon.stub(session, '_lookupAuthenticator').returns(authenticator);
+
+            let invalidation = session.invalidate().then((val) => {
+              expect(val).to.be.true;
+              done();
+            });
+
+            expect(invalidation instanceof RSVP.Promise).to.be.true;
+          });
```

Comment:
You can actually shorten this to

```js
it('it returns a resolved Promise', function(done) {
  return session.invalidate().then(() => {
    expect(true).to.be.true;
  });
 });
```

If you return a promise from a test, the runner will wait for that promise to resolve (and fail the test if it rejects). That way you don't need the `done`. Also the fact that the returned object is a promise is kind of implied in the use of `then` in `session.invalidate().then(…`.

Also I think you should not need to the stub here as that's only used after the early return anyway.